### PR TITLE
Servicenow small fixes

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/47762

## Description
- Added a type for the mirrored notes
- Fixed an issue while mirroring the assigned_to and the assignment_group fields

## Does it break backward compatibility?
   - [x] No

### CODE REVIEW

File: Packs/ServiceNow/ReleaseNotes/2_3_7.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+##### ServiceNow v2
+- Fixed an issue where the update of the *assigned_to* and *assignment_group* fields were updated only after another update was triggered.
+- Added the type for the mirroring notes (comments/work_notes).
```
Comment (ShahafBenYakir) [inline Packs/ServiceNow/ReleaseNotes/2_3_7.md:5]:
```suggestion
- Added the note type for the mirroring notes (comments/work_notes).
```

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 2]
```diff
@@ -2249,33 +2269,14 @@ def get_remote_data_command(client: Client, args: Dict[str, Any], params: Dict)
...
-    ticket['assigned_to'] = user_assigned
-
-    if caller:
-        user_result = client.get('sys_user', caller.get('value'))
-        user = user_result.get('result', {})
-        user_email = user.get('email')
-        ticket['caller_id'] = user_email
 
-    if ticket.get('resolved_by') or ticket.get('closed_at'):
+    if ticket.get('closed_at'):
```
Comment (ShahafBenYakir) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:2279]: This is a change we have not mentioned in the RN right?

**Review by ShahafBenYakir — COMMENTED** [review]
Looks good. Can we add a unit test that reproduces the bug you fixed?

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py (hunk 2 above)
Comment (Shellyber) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:2279]: Right.

**Comment by Shellyber** [comment]
> Looks good. Can we add a unit test that reproduces the bug you fixed?

Not really, since the issue here is the order of things.. can't really test that. The mirroring worked always, but for those specific fields, we needed to add comment to the ticket before we could see the change. 

I've also sent it to the customer for verification

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py (hunk 2 above)
Comment (ShahafBenYakir) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:2279]: Need to mention it in the RN as well

**Review by ShahafBenYakir — APPROVED** [review]

**Comment by ShahafBenYakir** [comment]
Just add the right RN

**Comment by Shellyber** [comment]
@ShahafBenYakir the customer reviewed and the fix provided is working.
Please review again

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 3]
```diff
@@ -2049,7 +2049,10 @@ def fetch_incidents(client: Client) -> list:
 
         severity = severity_map.get(result.get('severity', ''), 0)
 
-        parse_dict_ticket_fields(client, result)
+        try:
+            parse_dict_ticket_fields(client, result)
+        except Exception:
+            pass
```
[the remark this answers is not in the document]
Comment (Shellyber) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:2055]: No need, this is just a better handling for the fields and improves the customers experience with those fields in the mirror.

**Pull request merged**